### DISCUSSION TITLE
VS Code Auto-Import Not Using Custom Folder Aliases and #q-app Aliases

### DISCUSSION THREAD

**Discussion by julianctni** [quasarframework/quasar#17694] [General - Components / Directives / etc] [answered]
Hello,

last week I switcht my project to app-vite 2.0. Thanks again for the great work and docs!

However I encountered a small problem. In my project I am using a custom '@' alias, which points to 'src'. Before the update I added this alias to quasar.config > build and to the tsconfig file in my project root. After the update I added it only to the quasar.config file, since the tsconfig is now auto-generated according to the docs.

![image](https://github.com/user-attachments/assets/92839e44-8abf-4fff-9134-f2b062f6c24e)

The problem is now, that each time I want to auto-import something it ignores my custom alias and prefers the default 'src' alias from Quasar. I am guessing that happens because my custom alias is at the end of the path list in .quasar/tsconfig. Before the update, it always suggested the import with @ alias prefix.

Furthermore the #q-app/ aliases are also not suggested for auto-imports.

![image](https://github.com/user-attachments/assets/64fcf6b8-1e28-4100-b45a-edfe8ca6b182)

Is there anyone else who has this problem and knows a fix?

Best regards
Julian

**Comment by rstoenescu** [comment]
Just to be thorough: run the "quasar prepare" command and restart VSCode. Do you still have issues?

**Reply by julianctni** [reply]
I already tried this. Unfortunately the problem above still exists. When I manually move the custom @ alias to the top of the path list and restart VS Code, I get the auto import suggestions with '@'. However in the case of e.g. the defineBoot method, I stell get @quasar/app-vite/wrappers as auto-import path.

**Reply by rstoenescu** [reply]
Please confirm that your /tsconfig.json looks like this:

```
{
  "extends": "./.quasar/tsconfig.json"
}
```

Also, as a side note, `#q-app` is an alias to `@quasar/app-vite`, so the IDE suggestion is correct :)

**Reply by julianctni** [reply]
My /tsconfig.json looks like this:
```
{
  "extends": "./.quasar/tsconfig.json",
  "compilerOptions": {
    "types": ["vitest/globals"],
  }
}
```
> Also, as a side note, #q-app is an alias to @quasar/app-vite, so the IDE suggestion is correct :)

Yeah I know :) I am just wondering, why it isn't suggested

**Reply by rstoenescu** [reply, marked as answer]
It seems to prefer the actual package, unfortunately. Don’t think we can give it a hint for this. Nothing wrong with its suggestion though, but we recommend the alias, for future bulletproofing.

**Reply by julianctni** [reply]
Okay, seems so. But wouldn't it be possible to move the custom aliases defined in quasar.config file to the top of the path list? Then at least the custom ones would be prefered, if defined.
